parquet-column/src/main/java/parquet/schema/Types.java

```diff
@@ -555,6 +622,1024 @@ protected GroupType build(String name) {
+      return super.named(name);
+    }
+
+    public ListGroupElementBuilder<P, Q> groupElement(Type.Repetition repetition) {
+      Preconditions
+          .checkState(elementType == null, "groupElement can be called just once on a ListBuilder");
+      if (parent != null) {
+        return new ListGroupElementBuilder<P, Q>(this).repetition
+            (repetition);
+      } else {
+        return new ListGroupElementBuilder<P, Q>(this, returnType).repetition
+            (repetition);
+      }
+    }
+
+    public ListGroupElementBuilder<P, Q> requiredGroupElement() {
+      return groupElement(Type.Repetition.REQUIRED);
+    }
+
+    public ListGroupElementBuilder<P, Q> optionalGroupElement() {
+      return groupElement(Type.Repetition.OPTIONAL);
+    }
+
+    public ListElementBuilder<P, Q> element(PrimitiveTypeName type, Type.Repetition repetition) {
+      Preconditions
+          .checkState(elementType == null, "groupElement can be called just once on a ListBuilder");
+      if (parent != null) {
+        return new ListElementBuilder<P, Q>(this, type)
+            .repetition(repetition);
+      } else {
+        return new ListElementBuilder<P, Q>(this, returnType, type)
+            .repetition(repetition);
+      }
+    }
+
+    public ListElementBuilder<P, Q> requiredElement(PrimitiveTypeName type) {
+      return element(type, Type.Repetition.REQUIRED);
+    }
+
+    public ListElementBuilder<P, Q> optionalElement(PrimitiveTypeName type) {
+      return element(type, Type.Repetition.OPTIONAL);
+    }
+
+    public ListListElementBuilder<P, Q> listElement(Type.Repetition repetition) {
+      Preconditions
+          .checkState(elementType == null, "groupElement can be called just once on a ListBuilder");
+      if (parent != null) {
+        return new ListListElementBuilder<P, Q>(this).repetition(repetition);
+      } else {
+        return new ListListElementBuilder<P, Q>(this, returnType).repetition(repetition);
+      }
+    }
+
+    public ListListElementBuilder<P, Q> requiredListElement() {
+      return listElement(Type.Repetition.REQUIRED);
+    }
+
+    public ListListElementBuilder<P, Q> optionalListElement() {
+      return listElement(Type.Repetition.OPTIONAL);
+    }
+
+    public ListMapElementBuilder<P, Q> mapElement(Type.Repetition repetition) {
+      Preconditions
+          .checkState(elementType == null, "groupElement can be called just once on a ListBuilder");
+      if (parent != null) {
+        return new ListMapElementBuilder<P, Q>(this).repetition(repetition);
+      } else {
+        return new ListMapElementBuilder<P, Q>(this, returnType).repetition(repetition);
+      }
+    }
+
+    public ListMapElementBuilder<P, Q> requiredMapElement() {
+      return mapElement(Type.Repetition.REQUIRED);
+    }
+
+    public ListMapElementBuilder<P, Q> optionalMapElement() {
+      return mapElement(Type.Repetition.OPTIONAL);
+    }
+
+    public ListKeyElementBuilder<P, Q> keyElement(PrimitiveTypeName type,
```

Comment:
What is this class for? Do we need a ListKeyElement, and if so, do we need it here?